src/generators.rs

```diff
@@ -1561,10 +1614,11 @@ fn gen_fold(scale: u16) -> (Option<String>, String) {
         body.push_str(&statement);
     }
     println!("{}", body);
-    (None, body)
+
+    GenOutput::new(None, body, 1)
 }
 
-fn gen_map(scale: u16, input_size: u16) -> (Option<String>, String) {
+fn gen_map(scale: u16, input_size: u16) -> GenOutput {
```

Comment:
Actually, it might be nice to leave a comment above each generator function that takes in input size saying what the unit of the size is. For example in this case, it is arg size. 

I can open a pr to do that?